creativecommons/cc-legal-tools-app
# [Feature] Improve legal code language selection/fallback

## Fixes
- Related to [#448](https://github.com/creativecommons/cc-legal-tools-app/issues/448) by @TimidRobot

## Description
This pull request addresses the improvement of the legal code language selection and fallback logic in the `cc-legal-tools-app`. The current fallback mechanism for legal code translations uses a naive method, which has been updated to a more robust approach similar to the Deeds & UX translation logic.

The changes include:
1. Adding a new function `load_legal_code_translations()` to populate available legal code translations.
2. Replacing the `get_default_language_for_jurisdiction_naive()` function with `get_default_language_for_jurisdiction_legal_code()` for a more sophisticated fallback logic based on the newly added `settings.LANGUAGES_AVAILABLE_LEGAL_CODE`.
3. Renaming `get_default_language_for_jurisdiction_deed()` to `get_default_language_for_jurisdiction_deed_ux()` to clarify its scope.

## Technical details
- The `load_legal_code_translations()` function populates `settings.LANGUAGES_AVAILABLE_LEGAL_CODE` with the available languages for each jurisdiction.
- The fallback logic for legal code languages now checks whether the default language for a jurisdiction exists in the `settings.LANGUAGES_AVAILABLE_LEGAL_CODE`. If not, it falls back to the default language specified in the `settings.LANGUAGE_CODE`.

## Tests
To verify:
1. Ensure that the updated fallback logic for legal code languages works correctly based on the `settings.LANGUAGES_AVAILABLE_LEGAL_CODE`.
2. Test the Deeds & UX translations to confirm that the renaming of the fallback logic does not break existing behavior.




## Checklist



- [x ] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x ] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x ] My commit messages follow [best practices][best_practices].
- [x ] My code follows the established code style of the repository.
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin




For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
